datafusion/expr/src/logical_plan/extension.rs

```diff
@@ -30,6 +30,9 @@ pub trait UserDefinedLogicalNode: fmt::Debug + Send + Sync {
     /// Return a reference to self as Any, to support dynamic downcasting
     fn as_any(&self) -> &dyn Any;
 
+    /// Return the plan's name
```

Comment:
I ignored the `fmt_for_explain` method... You are right, they are redundant information.

>What would you think about making a function like the following that returned the result of format_for_explain

The returned string (like "TopK: k=1") is also not easy to use for the dispatching purpose, like we need to `.contains()` or `.start_with()` those strings. What do you think about separate name and parameters? E.g.:
```rust
fn name(&self) -> &'str {
  "TopK"
}

fn fmt_for_params(&self, f: &mut fmt::Formatter) -> fmt::Result { 
   write!(f, "k={}", self.k) 
}

// this can be provided as default implementation
fn fmt_for_explain(&self, f: &mut fmt::Formatter) -> fmt::Result { 
   write!(f, "{}: {}", self.name(), self.fmt_for_params()) 
}
```

Reply:
> What do you think about separate name and parameters? E.g.:

I think it would make sense, but could also be done as a follow on PR. I think this PR is good enough for now.

Thanks @waynexia